contracts/SimpleSwap.sol

```diff
@@ -78,23 +76,6 @@ contract SimpleSwap is SW3Utils {
     emit ChequeSubmitted(beneficiary, serial, amount);
   }
 
-  /// @notice submit a cheque
```

Comment:
* Whereas the idea to allow the owner to submit a cheque with a higher worth without the signature of the beneficiary is, in theory, an improvement, the current implementation of submitCheque does not achieve this since there is still a signature needed from the beneficiary for calling the smart contract. (you require the msg.sender to be the beneficiary)

* The actual overhead from the beneficiary to sign a cheque off-chain is negligible. Therefore, the improvement of being able to submit a cheque without a beneficiary signature is very little.

* Your concern of not allowing the owner to call the submitCheque function is valid; if you would do this he can indeed delay the payout indefinitely by submitting cheques with slightly higher worth

* The submitChequeLower function achieves exactly the same as the submitCheque function, only it can be sent to the blockchain by any party. Both the beneficiary and the owner must have signed it beforehand, as is the case with the submitCheque function too, although there the beneficiary signs by sending it to the blockchain.

* The major difference between the two is that you don't do a require in the submitChequeLower on the amount being bigger than a previously submitted amount (as is the case in the submitCheque function). I think, however, that there is no need to do this, as a beneficiary would not sign a cheque with a higher serial and lower amount if he would not really intend to pay with a waiver to the owner of the SWAP contract.

* You mention a TODO: security implications of anyone being able to call this and the resulting timeout delay above the function submitChequeLower. I don't see a security implication here.